tools/ant-garbage-collection.js

```diff
@@ -0,0 +1,144 @@
+const graphqlUrl = process.env.GRAPHQL_URL ?? 'https://arweave.net/graphql';
+
+const dryRun = process.argv.includes('--dry-run') ? true : false;
+const limit = process.argv.includes('--limit')
+  ? parseInt(process.argv[process.argv.indexOf('--limit') + 1])
+  : Infinity;
+
+const ao = connect({
+  CU_URL: cuUrl,
+  GRAPHQL_URL: graphqlUrl,
+});
+const ario = ARIO.init({
+  process: new AOProcess({
+    processId: arioProcessId,
+    ao,
+  }),
+  signer: createDataItemSigner(jwk),
+});
+
+const fetchAllArNSProcessIds = async () => {
+  const antIds = new Set();
+  let cursor = undefined;
+  let hasMore = true;
+
+  while (hasMore) {
+    console.log(`Fetching ANTs from cursor ${cursor}`);
+    const result = await ario.getArNSRecords({
+      cursor,
+      limit: 1000,
+    });
+    cursor = result.nextCursor;
+    hasMore = result.hasMore;
+    for (const item of result.items) {
+      antIds.add(item.processId);
+    }
+  }
+  console.log(`Found ${antIds.size} ANTs`);
+  return antIds;
+};
+
+const fetchAllProcessIdsInRegistry = async () => {
+  console.log('Fetching unregistered ants from registry...');
+  const antRegistryAntsRes = await ao.dryrun({
+    process: registryId,
+    From: vaotId,
+    Owner: vaotId,
+    data: "print(require('json').encode(require('.utils').keys(ANTS)))",
+    tags: [
+      {
+        name: 'Action',
+        value: 'Eval',
+      },
+    ],
+  });
+  const antRegistryAnts = JSON.parse(antRegistryAntsRes.Output.data);
+  console.log(`Found ${antRegistryAnts.length} ANTs in registry`);
+  return Array.from(antRegistryAnts);
+};
+
+async function main() {
+  console.log('Performing garbage collection on ANT registry...');
+
+  try {
+    const [processIdsForNames, processIdsInRegistry] = await Promise.all([
+      fetchAllArNSProcessIds(),
+      fetchAllProcessIdsInRegistry(),
+    ]);
+
+    const antsToUnregister = Array.from(processIdsInRegistry).filter(
+      (antId) => !Array.from(processIdsForNames).includes(antId),
+    );
+    console.log(`Found ${antsToUnregister.length} ANTs to unregister`);
+
+    function createUnregisterLuaTemplate(antIds) {
+      return `
+    Send({
+      Target = '${registryId}',
+      Action = 'Batch-Unregister',
+      Data = '${JSON.stringify(antIds)}',
+    })\n
+    `;
+    }
+
+    const unregisterLua = createUnregisterLuaTemplate(
+      antsToUnregister.slice(0, limit),
+    );
+
+    if (dryRun) {
+      console.log('Dry run enabled, skipping unregistration');
+      console.log('Unregister Lua preview:');
+      console.log(unregisterLua);
+      process.exit(0);
+    }
+
+    const vaotProcess = new AOProcess({
+      processId: vaotId,
+      ao,
+    });
+
+    const proposalRes = await vaotProcess.send({
+      tags: [
+        {
+          name: 'Action',
+          value: 'Propose',
+        },
+        { name: 'Proposal-Type', value: 'Eval' },
+        { name: 'Vote', value: 'yay' },
+        { name: 'Process-Id', value: registryId },
```

Comment:
This should actually be an eval on the _vaot_ - so that the message is sent from the vaot to the registry